asn1sccdocumentprocessor.cpp

```diff
@@ -23,83 +23,93 @@
 **
 ****************************************************************************/
 
-#include "documentprocessor.h"
+#include "asn1sccdocumentprocessor.h"
 
 #include <QFileInfo>
 
 #include "parseddocument.h"
+#include "asn1sccparseddocumentbuilder.h"
 
 using namespace Asn1Acn::Internal;
 
-DocumentProcessor::DocumentProcessor(const QString &projectName) :
-    m_projectName(projectName),
-    m_state(State::Unfinished),
-    m_docBuilder(nullptr)
+Asn1SccDocumentProcessor *Asn1SccDocumentProcessor::create(const QString &projectName)
 {
+    return new Asn1SccDocumentProcessor(projectName,
+                                        [](const QHash<QString, DocumentSourceInfo> &documents)->ParsedDocumentBuilder *
```

Comment:
this return type specification was required here? it looks strange with this space before `*` :D